spec/integration/user_index_spec.rb

```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+RSpec.describe 'Users page', type: :feature do
+  before(:all) do
+    visit users_path
+  end
+  describe 'display the user index page' do
+    it 'should have all usernames for all users' do
+      expect(page).to have_selector('h2', text: 'Tom')
+      expect(page).to have_selector('h2', text: 'Lilly')
+      expect(page).to have_selector('h2', text: 'Peter')
```

Comment:
- [ ] as part of the requirements, kindly add these missing tests for the user index : 

- Tests that asserts you can see the username of all other users.(currently you are testing only one user)
- Tests that asserts you can see the profile picture for each user.
- Tests that asserts you can see the number of posts each user has written.
- Tests that redirect to the user's show page when you click on a user.